### DISCUSSION TITLE
json_array_elements and json_build_object in Datafusion?

### DISCUSSION THREAD

**Discussion by dadepo** [apache/datafusion#6316] [Q&A] [answered]
From looking around the documentation, it looks like Datafusion does not natively have any functionality comparable to json_array_elements and json_build_object?

json_array_elements for allows arrays to be treated as rows
json_build_object for constructing json object from column values

See postgresql documentation for reference https://www.postgresql.org/docs/current/functions-json.html 

Is this the case? or there are ways to kinda achieve the same thing natively in Datafusion?

**Comment by Jefffrey** [comment, marked as answer]
For `json_build_object` we do now have `make_map` which seems to achieve something similar:

https://github.com/apache/datafusion/blob/48cc4c8af3a5ad500a44c8625ea44d6b4827af1e/datafusion/sqllogictest/test_files/map.slt#L132-L135

- https://datafusion.apache.org/user-guide/sql/scalar_functions.html#map

For `json_array_elements`, I don't think we have anything similar in DataFusion at the moment. I guess it would be classified as a table function, and you can see an example of implementing a custom user defined table function here:

https://github.com/apache/datafusion/blob/main/datafusion-examples/examples/udf/simple_udtf.rs
